certificate.go

```diff
@@ -0,0 +1,144 @@
+package letsencrypt
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// CertificateResponse holds response items after requesting a Certificate.
+type CertificateResponse struct {
+	Certificate *x509.Certificate
+	RetryAfter  int
+	URI         string
+	StableURI   string
+	Issuer      string
+}
+
+// Bundle bundles the certificate with the issuer certificate.
+func (c *CertificateResponse) Bundle() (bundledPEM []byte, err error) {
+	if !c.IsAvailable() {
+		return nil, errors.New("Cannot bundle without certificate")
+	}
+
+	if c.Issuer == "" {
+		return nil, errors.New("Could not bundle certificates. Issuer not found")
+	}
+
+	resp, err := http.Get(c.Issuer)
+	if err != nil {
+		return nil, fmt.Errorf("Error requesting issuer certificate: %s", err)
+	}
+
+	defer resp.Body.Close()
+	issuerDER, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading issuer certificate: %s", err)
+	}
+
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: c.Certificate.Raw})
+	issuerPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: issuerDER})
+
+	return append(certPEM, issuerPEM...), nil
+}
+
+// Retry request retries the certificate if it was unavailable when calling
+// NewCertificate or RenewCertificate.
+//
+// Note: If you are renewing a certificate, LetsEncrypt may return the same
+// certificate. You should load your current x509.Certificate and use the
+// Equal method to compare to the "new" certificate. If it's identical,
+// you'll need to start a new certificate flow.
+func (c *CertificateResponse) Retry() error {
+	if c.IsAvailable() {
+		return errors.New("Aborting retry request. Certificate is already available")
+	}
+
+	if c.URI == "" {
+		return errors.New("Could not make retry request. No URI available")
+	}
+
+	resp, err := http.Get(c.URI)
```

Comment:
I decided to make these on the CertificateResponse struct because that's the struct that has all the necessary info and retrying/bundling the response made more sense to me then retrying/bundling against the client. I'm happy to put in the client if you'd prefer that route -- passing in the CertificateResponse as a parameter would probably help with the readability of what you are retrying/bundling. I'll update.